upgrades/2024_10_09_032151_populate_case_started.php

```diff
@@ -0,0 +1,347 @@
+<?php
+
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Log;
+use ProcessMaker\Repositories\CaseUtils;
+use ProcessMaker\Upgrades\UpgradeMigration as Upgrade;
+
+class PopulateCaseStarted extends Upgrade
+{
+    const CHUNK_SIZE = 5000;
+
+    const REFRESH_TIME = 1;
+
+    private $lastPrint = 0;
+
+    /**
+     * Run any validations/pre-run checks to ensure the environment, settings,
+     * packages installed, etc. are right correct to run this upgrade.
+     *
+     * Throw a \RuntimeException if the conditions are *NOT* correct for this
+     * upgrade migration to run. If this is not a required upgrade, then it
+     * will be skipped. Otherwise the exception thrown will be caught, noted,
+     * and will prevent the remaining migrations from continuing to run.
+     *
+     * Returning void or null denotes the checks were successful.
+     *
+     * @return void
+     *
+     * @throws RuntimeException
+     */
+    public function preflightChecks()
+    {
+        //
+    }
+
+    /**
+     * Run the upgrade migration.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        $caseNumber = 1;
+        $chunkSize = self::CHUNK_SIZE;
+        $startTime = microtime(true);
+        echo '    Counting process requests...';
+        $count = $this->getNonSystemRequests()->count();
+        echo ' ', $count, PHP_EOL;
+
+        // Truncate the table cases_started
+        DB::table('cases_started')->truncate();
+        echo PHP_EOL;
+        echo '    Populating case_started from process_requests';
+
+        try {
+            $this->getNonSystemRequests()
+                ->orderBy('process_requests.id')
+                ->chunk($chunkSize, function ($requests) use (&$caseNumber, $startTime, $count) {
```

Comment:
Use the `CHUNK_SIZE` constant directly
```suggestion
                ->chunk(self::CHUNK_SIZE, function ($requests) use (&$caseNumber, $startTime, $count) {
```